testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java

```diff
@@ -527,17 +529,17 @@ protected void cleanEnvironment(KeycloakSession s) {
 
     @Before
     public final void createEnvironment() {
-        Time.setOffset(0);
+        setTimeOffset(0);
         USE_DEFAULT_FACTORY = isUseSameKeycloakSessionFactoryForAllThreads();
         KeycloakModelUtils.runJobInTransaction(getFactory(), this::createEnvironment);
     }
 
     @After
     public final void cleanEnvironment() {
-        Time.setOffset(0);
         if (getFactory() == null) {
             reinitializeKeycloakSessionFactory();
         }
+        setTimeOffset(0);
```

Comment:
Is it necessary to move the time offset after factory reinitialization?

Reply:
Yes, otherwise NPE is thrown.

Reply:
Can you share more details? What NPE?

Reply:
With `legacy-jpa+infinispan` profile:
```
java.lang.NullPointerException
	at org.keycloak.testsuite.model.KeycloakModelTest.checkProviderAvailability(KeycloakModelTest.java:156)
	at org.keycloak.testsuite.model.KeycloakModelTest.access$000(KeycloakModelTest.java:122)
	at org.keycloak.testsuite.model.KeycloakModelTest$1.lambda$apply$0(KeycloakModelTest.java:141)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.keycloak.testsuite.model.KeycloakModelTest$1.apply(KeycloakModelTest.java:143)

```

Reply:
this is failing because some tests are leaving the factory equal to `null` in that case I think it makes sense to move `setTimeOffset` after the initialization.